### PR TITLE
Tracker: yaw and pitch refactor

### PR DESCRIPTION
**Pull request by IamPete1**
This is abit of a tidy up of trackers handling of yaw and pitch in preparation to look at a possible yaw limits bug reported [here](https://discuss.ardupilot.org/t/tracker-1-1-officially-released/45576/4?u=iampete).

I have removed some filters that were used to try and tell where the center positions on yaw and pitch were, I'm not sure they would have ever worked as intended. 

I have also removed several variables that were passed to functions and not used.

There is alot of converting between earth and body frame that is quite hard to follow, maybe someone has a better understanding of this could take a look. 

I have not yet tested this (and I don't have the hardware to test all servo types) but it should be no functional change.

### CODE REVIEW

**Comment by IamPete1** [comment]
following a [update](https://discuss.ardupilot.org/t/tracker-1-1-officially-released/45576/10?u=iampete) from the user I have also reduced the default yaw range to 355

**Comment by lesaux** [comment]
I will definitely test this out in the upcoming days and will update here.

**Comment by peterbarker** [comment]
@lesaux Any news?

**Comment by lesaux** [comment]
TBH I mostly focused on https://github.com/ArduPilot/ardupilot/pull/12068.
As stated on the forum, I ran into issues when I ran this code, but I can try it again this weekend.
